src/app/clusters/concentration-measurement-server/concentration-measurement-server.h

```diff
@@ -0,0 +1,647 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "concentration-measurement-cluster-objects.h"
+#include <app/AttributeAccessInterface.h>
+#include <app/data-model/Nullable.h>
+#include <app/reporting/reporting.h>
+#include <app/util/af.h>
+#include <app/util/attribute-storage.h>
+#include <lib/support/IntrusiveList.h>
+#include <type_traits>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace ConcentrationMeasurement {
+
+struct DummyType
+{
+};
+
+/**
+ * This class provides the base implementation for the server side of the Concentration Measurement cluster as well as an API for
+ * setting the values of the attributes.
+ *
+ * @tparam NumericMeasurementEnabled whether the cluster supports numeric measurement
```

Comment:
What are "@tparam" @mhazley

Reply:
doxygen tag for Template Parameters, no?

Reply:
Ah, I did not know that!  Today is a good day already: I learned something.  ;)